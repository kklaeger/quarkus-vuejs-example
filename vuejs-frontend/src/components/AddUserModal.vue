<template>
  <form @submit.prevent="saveUser" class="cardwidth">
    <div class="modal-card" style="width: auto">
      <header class="modal-card-head">
        <p class="modal-card-title">Add User</p>
      </header>
      <section class="modal-card-body">
        <b-field label="Name">
          <b-input type="text" v-model="name" placeholder="Name" required />
        </b-field>
        <div v-if="this.$store.state.addUserIsLoading" class="loading">
          <LoadingIcon />
        </div>
        <div v-if="this.$store.state.addUserHasError" class="error">
          Unable to add user: {{ this.$store.state.addUserErrorMsg }}
        </div>
      </section>
      <footer class="modal-card-foot">
        <button class="button" type="button" @click="$emit('close')">
          Close
        </button>
        <button class="button is-primary" type="submit">
          Save
        </button>
      </footer>
    </div>
  </form>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import LoadingIcon from "@/components/LoadingIcon.vue";
@Component({
  components: { LoadingIcon }
})
export default class AddUserModal extends Vue {
  private name = "";

  private saveUser(): void {
    this.$store.dispatch("addUser", this.name).then(res => {
      if (res) this.$emit("close");
    });
  }
}
</script>

<style scoped lang="scss">
@import "./../styles/variables.scss";
.cardwidth {
  width: 20rem;
}
@media screen and (max-width: $tablet) {
  .cardwidth {
    width: 100%;
  }
}
.error {
  color: $error-message;
}
</style>
