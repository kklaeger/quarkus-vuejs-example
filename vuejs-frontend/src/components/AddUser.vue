<template>
  <div>
    <button class="button is-primary" @click="modalIsOpen = true">
      Add User
    </button>
    <div v-if="modalIsOpen">
      <b-modal active has-modal-card aria-role="dialog" aria-modal>
        <AddUserModal @close="modalIsOpen = false" />
      </b-modal>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import AddUserModal from "@/components/AddUserModal.vue";

@Component({
  components: { AddUserModal }
})
export default class AddUser extends Vue {
  private modalIsOpen = false;
}
</script>

<style scoped lang="scss"></style>
