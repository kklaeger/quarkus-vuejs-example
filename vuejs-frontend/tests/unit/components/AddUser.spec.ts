import { shallowMount, createLocalVue } from "@vue/test-utils";
import Buefy from "buefy";
import AddUser from "@/components/AddUser.vue";
import AddUserModal from "@/components/AddUserModal.vue";

const localVue = createLocalVue();
localVue.use(Buefy);

describe(AddUser.name + " component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should open and close modal.", async () => {
    const wrapper = shallowMount(AddUser, { localVue });

    expect(wrapper.findComponent(AddUserModal).exists()).toBeFalsy();

    await wrapper.find("button").trigger("click");

    expect(wrapper.findComponent(AddUserModal).exists()).toBeTruthy();

    await wrapper.findComponent(AddUserModal).vm.$emit("close");

    expect(wrapper.findComponent(AddUserModal).exists()).toBeFalsy();

    expect(wrapper).toMatchSnapshot();
  });
});
