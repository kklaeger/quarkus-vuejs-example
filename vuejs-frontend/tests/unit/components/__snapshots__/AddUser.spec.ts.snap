// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser component should open and close modal. 1`] = `
<div><button class="button is-primary">
    Add User
  </button>
  <!---->
</div>
`;
